examples/ui-react/src/components/MessageBubble.tsx

```diff
@@ -0,0 +1,95 @@
+import { Brain, Sparkles } from "lucide-react";
+import type { UIMessage, ReasoningMessage } from "@langchain/langgraph-sdk";
+
+/**
+ * MessageBubble component that renders human and AI text messages.
+ * Tool calls are handled separately by ToolCallCard.
+ */
+export function MessageBubble({ message }: { message: UIMessage }) {
+  const isHuman = message.type === "human";
+  const isSystem = message.type === "system";
+  const isReasoning = message.type === "reasoning";
+
+  // Extract text content from message
+  const getContent = () => {
+    if (typeof message.content === "string") {
+      return message.content;
+    }
+    if (Array.isArray(message.content)) {
+      return message.content
+        .filter((c): c is { type: "text"; text: string } => c.type === "text")
+        .map((c) => c.text)
+        .join("");
+    }
+    return "";
+  };
+
+  const content = getContent();
+
+  // Don't render if there's no content
+  if (!content) return null;
+
+  if (isReasoning) {
+    return <ReasoningBubble message={message} isStreaming={false} />;
+  }
+
+  return (
+    <div className="animate-fade-in">
```

Comment:
_nit_ it would be more readable if we didn't interleave styles for all kinds of messages